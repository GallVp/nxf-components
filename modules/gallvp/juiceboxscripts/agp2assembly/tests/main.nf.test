nextflow_process {

    name "Test Process JUICEBOXSCRIPTS_AGP2ASSEMBLY"
    script "../main.nf"
    process "JUICEBOXSCRIPTS_AGP2ASSEMBLY"

    tag "modules"
    tag "modules_gallvp"
    tag "juiceboxscripts"
    tag "juiceboxscripts/agp2assembly"

    test("HY - agp") {

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp')
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

    test("HY - agp - stub") {

        options '-stub'

        when {
            process {
                """
                input[0] = [
                    [ id:'test', single_end:false ], // meta map
                    file('https://github.com/Plant-Food-Research-Open/assemblyqc/raw/4f9fba392cf7212d61c6245fd72339e09164ce44/tests/hic/testdata/HYh1h2_Chr01_6000000_6015000.agp')
                ]
                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }

    }

}
