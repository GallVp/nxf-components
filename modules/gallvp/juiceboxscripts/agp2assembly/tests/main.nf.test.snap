{
    "HY - agp - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a108501cebf8df4cc42a0382ea8be68a"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a108501cebf8df4cc42a0382ea8be68a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-21T14:49:28.480672"
    },
    "HY - agp": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,48f07547635ccfd80eddc5b9a0d8040f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,a108501cebf8df4cc42a0382ea8be68a"
                ],
                "assembly": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test.assembly:md5,48f07547635ccfd80eddc5b9a0d8040f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,a108501cebf8df4cc42a0382ea8be68a"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.6"
        },
        "timestamp": "2025-07-21T14:49:24.507013"
    }
}